### ISSUE TITLE
in-game voice chat

### ISSUE THREAD

**Issue by HTPskittles** [schroeder-lvb/sissm#120]
Hey Schroeder!

First off, we love running sissm on our servers; thank you so much for all the hard work you've put into making a better gameplay experience for the rest of us!

Have you ever heard of sissm causing in-game voice chat (the built in one from NWI) not working on servers running sissm? We didn't have any issues prior to running sissm, but I personally doubt it's sissm causing the issue. I think it's coincidence that it functions randomly after restarting or closing sissm, but wanted to double check with you.

We're running 5 separate servers on Windows Server 2012. Each server has it's own folder, launch.bat, reboot.bat, sissm cfg, sissm exe, etc etc.  

Any input or knowledge of known issues is appreciated. Thanks man!

**Comment by schroeder-lvb** [comment]
Thank you for your feedback!

So far I have not received any other reports of broken voice services in association with SISSM -- I'll put out a APB (all-points-bulletin) on Discord if any other SISSM users are experiencing this issue.  I run 5 Linux servers using 3 VPS, instances and no issues.

SISSM uses RCON to communicate with the game server.  Last I checked, NWI has not given us an RCON command that allows SISSM to enable/disable voice services (in fact, I wish we had such a feature, so I can code up a SISSM !mute command!)

Couple of other leads to check out:

1.  Sandstorm uses Steam voice service  for in-game VoIP -- do you have firewall enabled and if so enough port range opened to cover all of your servers?

2.  Is it possible you have one or more instance of SISSM misconfigured to connect to a game port of same or another game server instead of RCON port?

3.  On Sandstorm 1.8.1 NWI changed something under-the-hood.  Many of us ran into Microsoft VCRUNTIME140 error that we did not see before the 1.8.1 release, and had to install (or re-install) the Microsoft runtime library manually.  Perhaps there's something going on here?

Some good tests to run:

1.  For the server(s) running SISSM with reproducible VoIP failures -- try running without SISSM to see if the problem persists.  I prefer you test it with only one game instance running "solo" just to rule out the possibility of cross-interaction/port allocation issue inter-instance.

2.  If you dare (!) try disabling your firewall temporarily to test if VoIP services are functional.

More details:

1.  I would like to know a little more about how you are running SISSM - what game mode, which plugins are active, which mods (mutators/maps) etc.   I'm aware of at least one third-party maps that is so out of compliance SISSM can't track the game state.

2.  Please make sure you are running the latest SISSM, currently: 0.2.8.

3.  Did the VoIP issue happen on a Tuesday (and only on Tuesday)?  Steam executes routine patch/reboot of their servers, and I wonder if their VoIP services were interrupted.

4.  Are you compiling SISSM from source, or using the pre-packaged binary?

**Comment by HTPskittles** [comment]
Like I said, I don't believe sissm to be the issue, it was suggested to me by another admin to bounce the idea off you anyway.

And it wasn't sissm! Yay! It was a combination of points 1 and 2 in your reply. 

When you mentioned ports it jogged my memory that, in my quest to resolve the in-game voice chat randomly breaking down, I changed the port range for all the servers because I figured the only logical reason for this issue was a port conflict. It was indeed a port conflict. 

We were using 27015-27030, 27036-27037 for everything. When I changed the scheme for the servers, I forgot about steam and just modified the rules to be 271xx... and didn't make new rules to accommodate steams requirements Haha

Thank you again for everything!
